### ISSUE TITLE
prefill - Add flag to include recently purchased games in download

### ISSUE THREAD

**Issue by tpill90** [tpill90/steam-lancache-prefill#365]
## Potential Implementation Behavior
* `select-apps`
  * A new column for the purchased date would be added to the selection grid view.  A potential column name could be **"Purchased"**.
  * A sort button to sort by purchase date would be added alongside the existing sort options, above the search bar.
* `prefill`
  * A new flag should be added which will add the recently purchased games to the download list.  A potential name for this flag could be something like `--recent-purchased`, however some thought should be given to a better name.  The naming should take care to be clear and distinct from the existing flag `--recent` which will include recently played games in the download list.
  * The new flag should include games purchased within a certain hard coded time period, such as days/weeks/months.  What is an appropriate time period is still to be determined.

## Screenshots
![image](https://github.com/user-attachments/assets/0662f4ee-53ac-4c70-ad0a-8f9ef52b4aeb)
![image](https://github.com/user-attachments/assets/5d0c6dc6-59b1-49a6-b0f7-3f69cc24fd54)

## Related issues
Related to #206 and #265

**Comment by Teddy-Moeller** [comment]
+1
**I am not a coder myself, so please take my suggestion for what it is**
Maybe it can be done with a script, that? (for Steam)
./SteamPrefill select-apps
->wait for login to complete<-
"CTRL"+"A" (To select all)
"ENTER" (To save)
"ARROW DOWN" (To not run now, in order to follow the cronjob configuration)
"ENTER" (To select)
------------------------------------
I have tried to do some google searches, and found that maybe the following solution can be of help:
* command "expect" (https://stackoverflow.com/questions/6264596/simulating-enter-keypress-in-bash-script)
* use pipes and/or redirection (https://superuser.com/questions/585398/sending-simulated-keystrokes-in-bash)
* ydootool (https://linuxcommandlibrary.com/man/ydotool)

I also came across "xdootool" but it seem like it need a graphical interface. So that is ruled out. ;)

As I stated above, I am not a coder, heck I do not even know how to make a fork and try my suggestions myself. But what I can do is to try and make it as easy for the ones that can... And I hope I have.

**Comment by tpill90** [comment]
Thanks for taking the time to think of some ideas on possible ways to implement this!  The one issue that I can see with what you have suggested is that this would always select every game on the account.  For a smaller library of curated games this probably won't be an issue.  But for a larger old library it could be, depending on how many game bundles have been purchased (gotta love the sale bundles), as well as the number of free to play games added.

I haven't had the time to look at this again, but I'd be more than happy to merge in changes if anyone else would be interested in implementing them.  I'm going to take a few minutes today to write up what functionality I think that the feature should have so that someone could have a clear target of what to implement.  I also have some existing work in progress code on a branch, so I'll make sure that people know that there is a starting point already.

**Comment by Treborbob** [comment]
Just wanted to add my support for this feature and share my use case, which an extra flag would addresses perfectly.

My internet connection is quite slow (~70Mbps), so prefilling games via Lancache is a huge help. Often, I'll buy a game on Steam via the mobile app while I'm away from home (e.g., at work). It would be ideal if my Lancache server could automatically start prefilling it soon after purchase so it's ready (or at least partially downloaded) by the time I get home.

My setup involves Lancache and the SteamPrefill container running in Docker on an Unraid server. Because the prefill process can saturate my connection (and there's no built-in throttling like the Steam client), I use an Unraid user script to run the prefill container only during off-peak hours (overnight).

Without a "recently purchased" option, I'd have to remember to manually run select-apps after buying a game to add it to the prefill list.

I've hacked something that works together on a fork of this project if someone wants to take a look -  I am a developer but never attempted to contribute to OSS before so I would need some guidance if what i've done looks like it's worth taking forward.

https://github.com/Treborbob/steam-lancache-prefill/tree/ft/new-purchases

**Comment by tpill90** [comment]
@Treborbob Youve been using this yourself after you made these changes and it's working correctly? I can try to take a look at the code today if you open a PR

**Comment by Treborbob** [comment]
@tpill90  Yep, i've used it, and it also still works with --recent, so you can combine them - also added extra output in the console when using --verbose. I'll open a PR, but as I said I am not totally familiar with contributing to public repo's so go easy!

**Comment by tpill90** [comment]
Reopening, not sure why it ended up getting closed.

**Comment by tpill90** [comment]
Released in [v3.5.0](https://github.com/tpill90/steam-lancache-prefill/releases/tag/v3.5.0)
